### DISCUSSION TITLE
Can you integrate Hekili into your Actionbars or rather your WeakAura?

### DISCUSSION THREAD

**Discussion by MarkusMoesbauer** [Hekili/hekili#775] [Q&A]
Hey guys,
so I came across the idea of how fantastic it would be to not have hekili in its own window, but just highlight (for example with a red border highlight) the recommended ability in your weakaura. By doing this I think it's way easier mostly for players new to their class to learn it and also I find it hard to use Hekili in PVP, but with that change I wouldn't have to watch both my weakaura and hekili.
So I was wondering if this is somehow possible? I guess for the actionbars, we would need some sort of addon update, but I could see it being possible in some custom weakaura (I am not really deep into WA, but if you guys say it is possible, I would look into it).

Greetings, 
m0si

**Comment by Hekili** [comment]
You can use the API to find out if a particular ability is the current recommendation:  `Hekili_GetRecommendedAbility( "Primary", 1 )` 

`"Primary"` refers to the Display -- you could use it for `"AOE"` or `"Interrupts"` or `"Defensives"` or `"Cooldowns"` as well.

`1` refers to the entry in the display; 1 is the first icon.

You'd probably want to tuck the addon display away somewhere, or set it to very low transparency (/hek > Displays > Primary > Visibility > Advanced; Alpha 0.01) for example.

**Reply by jwzarrin** [reply]
Sorry to necro this.  I just started with Hekili and I was overlaying it on my weakauras (I've only been using WA for a bit as well).  I love that Hekili displays the keybinds on each icon, and I was hoping to bring them over onto the WAs.  Hekili's generated images have the keybinds, but the ones generated from the weakaura call to the API do not have the keybind on it (just the image).

I wrote a Trigger (I basically copied and pasted from your other posts):

Type:  Custom
Event Type:  Status
Check On...:  Event(s)
Event(s):  HEKILI_RECOMMENDATION_UPDATE

Custom Trigger:

```
function ()
    if not Hekili then return false end

    return Hekili_GetRecommendedAbility( "Primary", 1 ) ~= nil
    end

```

Icon Info:

```
function ()
    if not Hekili then return false end

    return GetSpellTexture( Hekili_GetRecommendedAbility( "Primary", 1 ) )
end
```

It shows the updated icon appropriately, but it's just the ability that comes next - not the action bar reference (so no keybind on it).  Is there a way to adjust that?  I don't want to have to change the text display on each ability any time I change a keybind (I'd rather it would just automatically update whenever I use ElvUI to swap my keybinds).

I copy and pasted the weakaura code.  I don't know if this is helpful.


`!WA:2!TsvWUTTrq0OyyeuIIcxLc3KGGcwx0aBGIu7yGa0d9GKdvTATLuOORDrpqTK7qXTMA3f7U02YhkA0LMZcO9dqN7j9jKVacJ(fKpb)f0zjLCslAouGQdIZmC4mZBM3m1AvFuDAD6l)SzSybVVixfdp8w(XImH6Bxb)Pg3njrdMAZj84uHQNGXnr751jWZ)QyHiJkoN3)CMeg8QLQE0HqQIeBycU(P(AdrzE0uQimoxBeJsvO4WmX5PrjmotNUEHtAtuY4mXOydhck9h8i1cXxU2mkeLNKemwcQ99oOxRJoOPbvIQc38QhHLMWCzY1VwNhbNbCtF87yxmpCVg9dc7h0WpWNtgbANxNRHqsKwOIouqHb(LUhTpqYmPnZXAjsMrgdQQ4OLpCFVVR9bTd992R7Hh615zncA3Tt4r9qbVI7k)LKCEjGD3Clhx8hlXLlmU7dNYYyUMuG7QatUI7MqY0GlWPL(v(3I3u5C43agFiwmAe6dqBeH2mJ30DJEk2iIA8gFH7oUB5(ZFTlNLv(52ynrlHSS2uTZSQgsBCGk)1)pklSE6BdEaCbQdB(FRo3YbFTFCkeFAXdMy77227GBMr9uaoJ873Z7GdwoltzuW3Wgb0P58febh5hAPuNbbv6252FCRRr6iO4KSVhznio)XRpfazdSzeB8jiYtNs4y5y7bpT49ADfnxvQuYxuAS(5uDr9MCbhEfq0qFJc4dnP3zIvZY6k2P5icJ3Q4ll2UOEXofp5of7IpE))PfKPQr(hMaiDgIpFyODf4bV8JLFQbBEHL)Liuy9ekdTamuQGyMTYxz6nUSXNlLRvQPtj4g1ELRJ1QvBLxxzfYskxeB24OGUY7wAKKJDoeAXhZOihUbQEDv(yxa0sJV4QQL4FyXo9RkF)pH9CwY4IvNPe46dyhZn70TJxrDfccRpZkD8nxfMUaiCJCTwk2LUppNqrytCdcQc6ziMIYGbY6VfowM3fLXjlvlD5CHIESIiNC8cHRUjjLR(Dpk4G2D8w0yERRrfRk)K3v3TsRM8()loujuS1BYtF2LW9LB9UcgDmE8GfhAsvGofV1n(VHULWjkfydtnVOzMGq)9Pwgphz6Qbrgsgs4x3FuEMH54hNr061loXzgLH3PIrRJTQnXBNqPGglOklISfwqITvqDjYxXT9BptvsYSTOM2B4tVHCuSAZlfIr1MS8UC6mlzNvEw2r(rlnBN3pJPj44IMoprHhiXDaCyUYmJi(SQ9Q7vl832GrLR1NmsMblpGCCJimMjSHo(KmzkzLRRgmTSrXwtr93Z3ZRJ6IfnNNKZOZZ38R2(5xaz9usfJBZaKwqg4FEjjDoLPThFThWU9CgVAiG1W6Y7XgYfkORSedE2ZhEkLqPhCfs)X8U4no11(7(4DF821p7pp5Vo`
